sys/include/net/gnrc/ipv6/nib/conf.h

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (C) 2017 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_gnrc_ipv6_nib_conf  Configuration macros
+ * @ingroup     net_gnrc_ipv6_nib
+ * @brief       Configuration macros for network information base
+ * @{
+ *
+ * @file
+ * @brief       Configuration macro definitions for network information base
+ *
+ * @author      Martine Lenders <mlenders@inf.fu-berlin.de>
+ */
+#ifndef GNRC_IPV6_NIB_CONF_H_
+#define GNRC_IPV6_NIB_CONF_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @name    Compile flags
+ * @brief   Compile flags to (de-)activate certain features for NIB
+ * @{
+ */
+/**
+ * @brief   Use unique ID for DAD
+ *
+ * @see    [RFC 6775, section 6.5](https://tools.ietf.org/html/rfc6775#section-6.5)
+ */
+#ifndef GNRC_IPV6_NIB_HAS_UNIQUE_ID
+#define GNRC_IPV6_NIB_HAS_UNIQUE_ID         (0)
+#endif
+
+/**
+ * @brief   Always reverse translate link-local address
+ *
+ * @see [RFC 6775, section 5.6](https://tools.ietf.org/html/rfc6775#section-5.6)
+ */
+#ifndef GNRC_IPV6_NIB_ALWAYS_REV_TRANS
+#define GNRC_IPV6_NIB_ALWAYS_REV_TRANS      (0)
+#endif
+
+/**
+ * @brief   Queue packets for waiting to complete address registration
+ *
+ * @see [RFC 4861, section 5.1](https://tools.ietf.org/html/rfc4861#section-5.1)
+ * @see [RFC 6775, section 5.6](https://tools.ietf.org/html/rfc6775#section-5.6)
+ */
+#ifndef GNRC_IPV6_NIB_QUEUE_PACKETS
+#define GNRC_IPV6_NIB_QUEUE_PACKETS         (1)
+#endif
+
+#ifndef GNRC_IPV6_NIB_DO_HANDLE_NBR_SOL
```

Comment:
They are used as compile switches, but I did not want them to be `#ifndef` enabled flags, but `#if` enabled flags, so you can *deactivate* them beyond the default config easily, too. This is actually more important than *activating* them, since e.g. 6Lo-ND doesn't need to hantle TLLAOs or redirect messages.